vaas/client.go

```diff
@@ -184,6 +202,28 @@ func (c *defaultClient) GetDC(name string) (*DC, error) {
 	return nil, fmt.Errorf("no DC with name %s found", name)
 }
 
+func (c *defaultClient) FindBackendID(director string, address string, port int) (int, error) {
+	directorFound, err := c.FindDirector(director)
+	if err != nil {
+		return 0, fmt.Errorf("cannot determine director ID: %s", err)
+	}
+	for _, backendURL := range directorFound.BackendURLs {
```

Comment:
I have problem with this bock. In case of aplication restart, this could generate large number of errors. Because if you have 100 replicas in the cluster and you allow to restart 100 of them at a time, and if you fetch every backend on the list there, often will be in a situation when it is simply deleted because other replica took it down. 

Not to mention you making one call for each backend, it might be time consuming if pod has a lot of replicas, can you imagine listing 100 replicas this way * times 100 because this many replicas are restarting/deploying? And at the top of that, deregistering will be simply very slow.

I suggest to list all backends in one request, find matching port and ip on the list, if you found it fetch backend associated Director to confirm its name or something like that.

Reply:
Fixed it, VaaS has an undocumented search feature.